tests/plugins/test_deepspeed_plugin.py

```diff
@@ -649,26 +650,40 @@ def test_deepspeed_multigpu_stage_2_accumulated_grad_batches(tmpdir, offload_opt
 
     class VerificationCallback(Callback):
 
+        def __init__(self):
+            self.on_train_batch_start_called = False
+            self.on_before_backward_called = False
+
         def on_train_batch_start(
             self, trainer, pl_module: LightningModule, batch: Any, batch_idx: int, dataloader_idx: int
         ) -> None:
             deepspeed_engine = trainer.training_type_plugin.model
             assert trainer.global_step == deepspeed_engine.global_steps
+            self.on_train_batch_start_called = True
+
+        def on_before_backward(self, trainer, pl_module, loss):
```

Comment:
is this something that you specifically need for deepspeed?

Reply:
No, just a test to make sure it is being called properly.

Reply:
@tchaton why is this in the deepspeed test? Should it not be in it's own general test?